components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -85,17 +87,31 @@
     private static String updateServiceUrl(String currentUrl) throws ResolverException {
         /*
             Supported formats
-            https://{host}:{port}/api1
-            https://{url}/api1
+            https://{MI_HOST}:{MI_PORT}/api1
+            https://{MI_URL}/api1
         */
-        SystemResolver resolver = new SystemResolver();
-        if (currentUrl.contains(HOST) && currentUrl.contains(PORT)) {
-            resolver.setVariable(MI_HOST);
-            String host = resolver.resolve();
-            resolver.setVariable(MI_PORT);
-            String port = resolver.resolve();
-            currentUrl = currentUrl.replace(HOST, host).replace(PORT, port);
+        if (currentUrl.contains(HOST) || currentUrl.contains(PORT)) {
+            Map<String, Object> configParser = ConfigParser.getParsedConfigs();
+            String envHost = System.getenv(MI_HOST);
+            // update the {MI_HOST} with server.hostname value
+            if (currentUrl.contains(HOST) && (envHost == null || envHost.isEmpty())) {
+                envHost = (String) configParser.get(SERVER_HOSTNAME);
```

Comment:
Shall we get the hostname as in https://github.com/wso2/micro-integrator/blob/57715992edaacfd50a40036baa2b4f467307e218/components/org.wso2.micro.integrator.core/src/main/java/org/wso2/micro/integrator/core/internal/CoreServerInitializer.java#L488 ?
If it makes more sense, we can move this to a common method in MicroIntegratorBaseUtils

Reply:
Fixed